util/modbus/modbus.go

```diff
@@ -375,14 +375,26 @@ func RegisterOperation(r Register) (rs485.Operation, error) {
 		op.FuncCode = modbus.FuncCodeReadHoldingRegisters
 	case "input":
 		op.FuncCode = modbus.FuncCodeReadInputRegisters
+	case "coil":
+		op.FuncCode = modbus.FuncCodeReadCoils
+		r.Decode = "bool1"
 	case "writesingle":
 		op.FuncCode = modbus.FuncCodeWriteSingleRegister
+	case "writesinglecoil":
+		op.FuncCode = modbus.FuncCodeWriteSingleCoil
+		r.Decode = "bool1"
 	default:
 		return rs485.Operation{}, fmt.Errorf("invalid register type: %s", r.Type)
 	}
 
 	switch strings.ToLower(r.Decode) {
 
+	// 1 bit (coil)
+	case "bool":
+		fallthrough
+	case "bool1":
+		op.Transform = decodeBool1()
```

Comment:
```suggestion
		op.Transform = decodeBool1
```

und oben die Verschachtelung raus.

Reply:
ok